### ISSUE TITLE
Transparent area of image shows  light shade when Image Filter applied.

### ISSUE THREAD

**Issue by mk10391** [fabricjs/fabric.js#6508]
## Version
3.6.3

## Information about environment
browser?
IE11

## Steps to reproduce

if i add an image with transparent area to canvas (canvas should have some dark color or place your image on top of another dark image ) and apply contrast filter, you will notice that the transparent area gets a light shade (also noticed with vintage & noise filter but with contrast filter its very visible.) . i think it has started happening in the last few weeks as it was not happening before. it happens with IE11 but things just work fine in Chrome. 

while googling i found some old link but probably things are different with fabric js version 3.6.3 
https://stackoverflow.com/questions/50017012/transparency-issue-on-ie-with-blendcolor-filter

here are the basic steps.
- Add an image to the canvas with Transparent background.
- Apply filter  contrast filter to image

## Expected Behavior
the transparent area of the image show light shade in IE11. ....but it works perfectly in chrome.

## Actual Behavior
the transparent area of the image should not be impacted in IE when applying filter.

**Comment by mk10391** [comment]
![image](https://user-images.githubusercontent.com/63499328/89727741-3cdfd500-d9ed-11ea-9f36-df06fd3f0f1d.png)

**Comment by mk10391** [comment]
you see that light area around star. that is a result of Contrast filter. that light area is transparent.

**Comment by asturur** [comment]
Can you share a working jsfiddle?

**Comment by asturur** [comment]
oh it looks like the stackoverflow answer is mine and was investigated
already.

**Comment by mk10391** [comment]
ok so how to fix it ? that piece of code in the stackoverflow is not there in the fabricjs version 3.6.3....so i am not sure what to replace.  But all this thing was working fine till recently in IE11 just few weeks back and now IE11 shows that shade.

**Comment by asturur** [comment]
may be hardware dependant or driver dependant. Did you change dev machine?

**Comment by mk10391** [comment]
no i did not change anything. i am using the same code and same VS2019 .net core and same windows 8.1 pro laptop. i restarted my PC thinking it might be something on my end and its still the same. happening only if i run it through IE11.

**Comment by mk10391** [comment]
![image](https://user-images.githubusercontent.com/63499328/90985011-8af0ee80-e53e-11ea-811e-77dca8e727c6.png)

i tried to use the condition mentioned as a solution.....it removes the main shade but it still leaves light lines on the image. as you see on my pasted image (unless i make it really dark which i don't want).

**Comment by mk10391** [comment]
in chrome, applying contrast doesn't leave any light lines. below is my result from Chrome (looks perfect in chrome)
![image](https://user-images.githubusercontent.com/63499328/90985138-8973f600-e53f-11ea-87f7-d2b845c40a54.png)

so IE11 still not showing the proper contrast (shows light lines).....any idea ?

**Comment by asturur** [comment]
yes we need to fix ALL the filters to take care of ie11. Mark the code as ie11 only so that it can be removed when we stop supporting it.

**Comment by mk10391** [comment]
Just wanted to Update here that i just tested this issue with fabric js 4.1.0 and the issue exist in this version as well for IE11. i applied the fix for vintage/contrast/noise filters but it still leaves  a light line just as i mentioned above.
